core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -172,10 +172,10 @@ void *impl_allocate_common(const Cuda &exec_space, const char *arg_label,
     if (exec_space_provided) {
       cudaStream_t stream = exec_space.cuda_stream();
       error_code          = cudaMallocAsync(&ptr, arg_alloc_size, stream);
-      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamSynchronize(stream));
+      exec_space.fence("Kokkos::Cuda: backend fence after async malloc");
```

Comment:
Note that this case will need to be tested manually, since `KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC` is defaulted off in the configuration due to #4228

Reply:
I think it looks right, though, and makes things closer to what @tcclevenger is doing in #5989. Hopefully this doesn't create a conflict.

Reply:
> Note that this case will need to be tested manually, since `KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC` is defaulted off in the configuration due to #4228

- [ ] I'll test it. We are using `KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC` in our builds ;)

Reply:
BTW, can you explain how the actions will test if I broke something on Cuda? I only see `OpenMP` and `HPX` in the checks for now...

Reply:
Click on the `jenkin-ci`'s details. It looks like you didn't run clang-format-8: https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/12572/pipeline.

Reply:
"Open Blue Ocean" tab in jenkin-ci will show all the configs once clang format passes.

Reply:
@PhilMiller If you agree, I'll close this thread and keep the changes.